### ISSUE TITLE
Lagging on playtime

### ISSUE THREAD

**Issue by xasans** [vi3itor/GameplayFootball#5]
Hi @vi3itor  i build project for windows but game is lagging on the playing match. Is it normal?

**Comment by vi3itor** [comment]
Hi! 
No, it shouldn't lag unless you have a low-spec PC or running it in the virtual machine. The other reason might be Debug build: on one of the machines, when I used `cmake`, that was installed with Visual Studio, it built Debug version even when I passed the `--Release` flag. It was solved by installing [the latest version of CMake](https://cmake.org/download/).
Can you try running the initial version of the game, compiled and published by Bastiaan?

**Comment by vi3itor** [comment]
You can find it attached. I had to compress `rar` archive into `zip`, because GitHub doesn't support attaching `rar`-files. 
[gameplayfootball_beta2v02_windows.zip](https://github.com/vi3itor/GameplayFootball/files/6883484/gameplayfootball_beta2v02_windows.zip)

**Comment by acoppola** [comment]
Another reason is that the game choose automatically the integrated graphics if you are on a system that have an integrated and a discrete videocard (some laptops and PCs).
For nVidia users go to the control panel and add your executable to the list and in order to run the game with graphic card.

**Comment by vi3itor** [comment]
Thank you Antonio for the valuable comment! I will add it to the installation instructions.

**Comment by xasans** [comment]
@vi3itor  Yes i have tried compiled version by Bastiaan and works good. 
@acoppola  Yes my videocard is integrated and i add game to control panel of videocard but lagging doesn't stop
My CPU Intel Core i3-10100 with Intel UHD 630 videocard

**Comment by acoppola** [comment]
> 630

if you have only an integrated videocard performance isn't good. If you have an integrated and a discrete videocard (like me, Intel integrated and GeForce 840M) you should add the exe path in nvidia control panel to use the latter.

**Comment by xasans** [comment]
OK @acoppola but why does version of Batsiaan plays fine on my pc?

**Comment by acoppola** [comment]
@xasans dunno, can be that Bastian one use SDL 1.2, or it was compiled in a different way, or other stuff (there were a lot of changes). I had slowdowns and I solved that way.

**Comment by xasans** [comment]
@acoppola  So did you compile on windows or other OS?

**Comment by vi3itor** [comment]
@xasans Try applying @WesleyKhoiVo's pull request #2 and please report your CPU usage. I will also publish the binaries tomorrow so you can test current version on your system.
I am also planning to disable logging, which should speed up the game.

**Comment by VOM7HC** [comment]
I just improve it minor.

But I think we must change mathematics formula so much from float to integer. Because Bastiaan used too much float - double math, but CPU is better calculate integer.

For example:
number * 0.5 = half number  => number / 2 =  half number (if it just 2 - integer, no need cast number to float like 0.5)

Useful link:
[Floating point unit](https://en.wikipedia.org/wiki/Floating-point_unit)
[FLOPS](https://en.wikipedia.org/wiki/FLOPS)

**Comment by VOM7HC** [comment]
One of reasons I think Bastiaan's compiled version working better is MSVC and Mingw. Now Vi3itor version is built from MSVC and Bastiaan is Mingw, that why his code can use some VLAs. 

I don't say we should be compile on Mingw and VLAs (which is bad code pattern - not thread safe from C99 not C++) and a lot of AAA game for Windows is compiled with MSVC instead of MingW, but may be this is the reason why performance is dropped (since he has optimized his code on MingW instead MSVC).

[Why aren't VLAs in C++](https://stackoverflow.com/questions/1887097/why-arent-variable-length-arrays-part-of-the-c-standard)

**Comment by vi3itor** [comment]
Yeah, it would be interesting to see to what extent VLAs affect the performance.

**Comment by VOM7HC** [comment]
> Yeah, it would be interesting to see to what extent VLAs affect the performance.

Yeah, Array allocated in Stack and Vector is allocated in both Heap and Stack (although it was introduced as alternative Array).

[When vectors are allocated, do they use memory on the heap or the stack?](https://stackoverflow.com/questions/8036474/when-vectors-are-allocated-do-they-use-memory-on-the-heap-or-the-stack)

**Comment by vi3itor** [comment]
@xasans I have published the binaries, you can [download](https://github.com/vi3itor/GameplayFootball/releases/download/beta4/gameplayfootball_beta4v04_windows.zip) and run it on your machine.

An important question: have you tried to decrease the screen resolution? In the binaries above, I set 1280x720 in the config by default. You can try other resolutions supported by your machine (Settings -> Graphics).

**Comment by VOM7HC** [comment]
Hello all, I create a discussion, topic: improve performance, so anyone care performance can join this discussion. #6

**Comment by WesleyKhoiVo** [comment]
Hello all, I create a discussion, topic: improve performance, so anyone care performance can join this discussion. #6

**Comment by ESC4P1STS** [comment]
> Can you try running the initial version of the game, compiled and published by Bastiaan?

Hello I tried to run the demo but that didn't work then I downloaded the source code and used the ReadMe.md to compile and it still didn't work, I use windows but I didn't find the gameplayfootball.exe at the end of all that.

**Comment by WesleyKhoiVo** [comment]
`I downloaded the source code and used the ReadMe.md to compile and it still didn't work`
Did you switch to Windows branch before compile?
Could you give me more detail about your issue (log build or crash or your machine details) ?

**Comment by VOM7HC** [comment]
> > Can you try running the initial version of the game, compiled and published by Bastiaan?
> 
> Hello I tried to run the demo but that didn't work then I downloaded the source code and used the ReadMe.md to compile and it still didn't work, I use windows but I didn't find the gameplayfootball.exe at the end of all that.

Did you switch to Windows branch before compile?
Could you give me more detail about your issue (log build or crash and your machine information)?

**Comment by ESC4P1STS** [comment]
> Did you switch to Windows branch before compile?
> Could you give me more detail about your issue (log build or crash and your machine information)?

Yes and I just checked again and it said I have already compiled.
I got to the build section for compiling on windows and this passage of information was confusing asking me to " Make sure that you correctly set the directory for vcpkg (in our case it is installed into C:\dev\vcpkg):" at this moment in time I am still using the Gameplay Football directory to make the build so I don't know whether I messed up there since everything else was fine. At the end of all of this I couldn't build the Debug version giving me an error at the end. I don't know whether you understand me so I might try and take pictures granted I will continue where I left off

**Comment by VOM7HC** [comment]
Have you ever tried my new instruction build on Windows with VS 2019? This build is support Debugging mode. Since debug mode it will generate text log in build folder when game is running

**Comment by ESC4P1STS** [comment]
>  my new instruction build on Windows with VS 2019?

no I haven't I'll try and use it should i restart the build

**Comment by DoomDivine** [comment]
> Thank you Antonio for the valuable comment! I will add it to the installation instructions.

Hey there, i had a doubt regarding this game:

So, if i want to create a player, for example a football player with an overall rating of 999 or 1 million ovr, such that he scores from goal to goal and cannot be tackled and is incredibly fast, how would i do that in the game's sources code? How to change the player overall in the game's source code? Please tell me how do i do this, because this has always been my dream modification in a football game, sadly fifa does not allow me to make 1 million ovr players, but i hope this game allows me? Please help me out?

**Comment by DoomDivine** [comment]
> Hi! No, it shouldn't lag unless you have a low-spec PC or running it in the virtual machine. The other reason might be Debug build: on one of the machines, when I used `cmake`, that was installed with Visual Studio, it built Debug version even when I passed the `--Release` flag. It was solved by installing [the latest version of CMake](https://cmake.org/download/). Can you try running the initial version of the game, compiled and published by Bastiaan?

Hey there, i had a doubt regarding this game:

So, if i want to create a player, for example a football player with an overall rating of 999 or 1 million ovr, such that he scores from goal to goal and cannot be tackled and is incredibly fast, how would i do that in the game's sources code? How to change the player overall in the game's source code? Please tell me how do i do this, because this has always been my dream modification in a football game, sadly fifa does not allow me to make 1 million ovr players, but i hope this game allows me? Please help me out?
